diff --git a/classes/swupdate-common.bbclass b/classes/swupdate-common.bbclass
index 6310ce5409c9..bfc88f2800b0 100644
--- a/classes/swupdate-common.bbclass
+++ b/classes/swupdate-common.bbclass
@@ -225,7 +225,7 @@ def swupdate_add_src_uri(d, list_for_cpio):
         aes_file = d.getVar('SWUPDATE_AES_FILE', True)
         if aes_file:
             key,iv = swupdate_extract_keys(d.getVar('SWUPDATE_AES_FILE', True))
-        if (filename != 'sw-description') and (os.path.isfile(local)):
+        if (filename != 'sw-description') and (filename != 'otab-shell') and (os.path.isfile(local)):
             encrypted = (d.getVarFlag("SWUPDATE_IMAGES_ENCRYPTED", filename, True) or "")
             dst = os.path.join(s, "%s" % filename )
             if encrypted == '1':
@@ -301,14 +301,16 @@ def swupdate_create_cpio(d, swudeploydir, list_for_cpio):
 python do_swuimage () {
     import shutil
 
-    list_for_cpio = ["sw-description"]
+    # Add sw-description and otab-shell files to the swu file from WORKDIR
+    list_for_cpio = ["sw-description", "otab-shell"]
     workdir = d.getVar('WORKDIR', True)
     s = d.getVar('S', True)
     imgdeploydir = d.getVar('SWUDEPLOYDIR', True)
     shutil.copyfile(os.path.join(workdir, "sw-description"), os.path.join(s, "sw-description"))
+    shutil.copyfile(os.path.join(workdir, "otab-shell"), os.path.join(s, "otab-shell"))
 
     if d.getVar('SWUPDATE_SIGNING', True):
-        list_for_cpio.append('sw-description.sig')
+        list_for_cpio.insert(1, 'sw-description.sig')
 
     # Add artifacts added via SRC_URI
     if not d.getVar('INHIBIT_SWUPDATE_ADD_SRC_URI', True):
